resources/lta_cmd.py

```diff
@@ -0,0 +1,205 @@
+"""
+Command line utility for Long Term Archive.
+
+Run with `python -m resources.lta_cmd $@`.
+"""
+
+import argparse
+import asyncio
+import json
+import sys
+from typing import Dict
+
+from rest_tools.client import RestClient  # type: ignore
+
+from lta.config import from_environment
+
+EXPECTED_CONFIG = {
+    'LTA_REST_TOKEN': None,
+    'LTA_REST_URL': None
+}
+
+
+async def display_config(args) -> int:
+    if args.json:
+        print_dict_as_pretty_json(args.config)
+    else:
+        for key in args.config:
+            print(f"{key}:\t\t{args.config[key]}")
+    return 0
+
+
+def print_dict_as_pretty_json(d: Dict) -> None:
+    print(json.dumps(d, indent=4, sort_keys=True))
+
+
+async def request_ls(args) -> int:
+    rc = RestClient(args.config["LTA_REST_URL"], token=args.config["LTA_REST_TOKEN"])
+    try:
+        response = await rc.request("GET", "/TransferRequests")
+        if args.json:
+            print_dict_as_pretty_json(response)
+        else:
+            results = response["results"]
+            for x in results:
+                short_id = x["uuid"][:7]
+                create_time = x["create_timestamp"]
+                source_cook = x["source"].split(":")
+                path = source_cook[1]
+                source = source_cook[0]
+                dests = [y.split(":")[0] for y in x["dest"]]
+                print(f"{short_id}\t{create_time}\t{path}\t{source} -> {dests}")
+    except Exception as e:
+        print(e)
+        return -1
+    return 0
+
+
+async def request_new(args) -> int:
+    # configure a RestClient from the environment
+    rc = RestClient(args.config["LTA_REST_URL"], token=args.config["LTA_REST_TOKEN"])
```

Comment:
The RestClient configuration smells like duplicate code, and should be factored out somewhere.  Some options:
* an external setup function
* making a class for all the functions, and doing "setup" stuff in the `__init__`
or something else if you can think of anything better.

Reply:
Refactored this so that RestClient is configured in main and passed along via `set_defaults`.